qutebrowser/browser/hints.py

```diff
@@ -570,23 +570,31 @@ def _check_args(self, target, *args):
                 raise cmdutils.CommandError(
                     "'args' is only allowed with target userscript/spawn.")
 
-    def _filter_matches(self, filterstr, elemstr):
-        """Return True if `filterstr` matches `elemstr`."""
+    def _filter_matches(self, filterstr: str, elem):
+        """Return True if `filterstr` matches `elem` text or any of the HTML attributes
+        from the `hints.extended_filter_attrs` configuration list."""
         # Empty string and None always match
         if not filterstr:
             return True
         filterstr = filterstr.casefold()
-        elemstr = elemstr.casefold()
-        # Do multi-word matching
-        return all(word in elemstr for word in filterstr.split())
+        elemstr = str(elem).casefold()
 
-    def _filter_matches_exactly(self, filterstr, elemstr):
-        """Return True if `filterstr` exactly matches `elemstr`."""
+        matchagainst = elemstr
+        for _attr in config.val.hints.extended_filter_attrs:
```

Comment:
Can you use `config.cache['hints.extended_filter_attrs']` here instead? Currently it will do a loop over every char in every string in the array (which is called on every hint) which can be costly